### PR TITLE
Update remote docs for 3.7.0

### LINKED ISSUE #345: Document singularity remote enhancements and implications

### PR DESCRIPTION
**Pull request by ikaneshiro**
## This fixes or addresses the following GitHub issues:

- Fixes #345

### CODE REVIEW

File: endpoint.rst [hunk 1]
```diff
@@ -6,40 +6,24 @@ Remote Endpoints
 Overview
 --------
 
+The ``remote`` command group allows users to manage the service endpoints
+Singularity will interact with for many common command flows. This includes
+managing credentials for image storage services, remote builders, and key 
+servers used to locate public keys for SIF image verification. Currently,
+there are three main types of remote endpoints managed by this command group:
+the public Sylabs Cloud, OCI registries and keyservers.
```
Comment (dtrudg) [inline endpoint.rst]:
```suggestion
the public Sylabs Cloud (or local Singularity Enterprise installation), OCI registries and keyservers.
```

File: endpoint.rst [hunk 2]
```diff
@@ -229,30 +229,303 @@ as ``push``, ``pull`` etc. via ``remote use``:
...
+    Keyservers
+    ==========
+
+    URI                       GLOBAL  INSECURE  ORDER
+    https://keys.example.com  YES     NO        1*
+
+    * Active cloud services keyserver
+
+
+Singularity 3.7 introduces the ability for an administrator to make a remote
```
Comment (dtrudg) [inline endpoint.rst:275]: Probably worth mentioning this in the admin docs

File: endpoint.rst [hunk 3]
```diff
@@ -229,30 +229,303 @@ as ``push``, ``pull`` etc. via ``remote use``:
...
+    https://pgp.example.com  YES     NO        1
+    https://keys.sylabs.io   YES     NO        2*
+
+    * Active cloud services keyserver
+
+Since we specified ``--order 1``, the ``https://pgp.example.com`` keyserver was
+placed as the first entry in the list and the default keyserver was moved to
+second in the list. With the keyserver configuration above, all image default
+image verification performed by Singularity will first reach out to
+``https://pgp.example.com`` and then to ``https://pgp.example.com`` when
```
Comment (dtrudg) [inline endpoint.rst]:
```suggestion
``https://pgp.example.com`` and then to ``https://keys.sylabs.io`` when
```

File: endpoint.rst [hunk 4]
```diff
@@ -229,30 +229,303 @@ as ``push``, ``pull`` etc. via ``remote use``:
...
+    URI                      GLOBAL  INSECURE  ORDER
+    https://pgp.example.com  YES     NO        1
+    https://keys.sylabs.io   YES     NO        2*
+
+    * Active cloud services keyserver
+
+    Authenticated Logins
+    =================================
+
+    URI                     INSECURE
```
Comment (dtrudg) [inline endpoint.rst:432]: Quick mention of what `INSECURE` means?

File: endpoint.rst [hunk 5]
```diff
@@ -229,30 +229,303 @@ as ``push``, ``pull`` etc. via ``remote use``:
...
+images or the registry itself. Previously, the only methods in Singularity to
+supply credentials to registries were to supply credentials for each command or
+set environment variables for a single registry.
+See :ref:`Authentication via Interactive Login <sec:authentication_via_docker_login>`
+and :ref:`Authentication via Environment Variables <sec:authentication_via_environment_variables>`
+
+Singularity 3.7 introduces the ability for users to supply credentials on a per
+registry basis with the ``remote`` command group.
+
+Users can log into an oci registry with the ``remote login`` command by
```
Comment (dtrudg) [inline endpoint.rst]:
```suggestion
Users can login to an oci registry with the ``remote login`` command by
```

File: endpoint.rst [hunk 6]
```diff
@@ -229,30 +229,303 @@ as ``push``, ``pull`` etc. via ``remote use``:
...
+
+    URI                     GLOBAL  INSECURE  ORDER
+    https://keys.sylabs.io  YES     NO        1*
+
+    * Active cloud services keyserver
+
+    Authenticated Logins
+    =================================
+
+    URI                 INSECURE
```
Comment (dtrudg) [inline endpoint.rst:483]: Need to explain `INSECURE` somewhere.

File: endpoint.rst [hunk 7]
```diff
@@ -229,30 +229,303 @@ as ``push``, ``pull`` etc. via ``remote use``:
...
+    docker://docker.io  NO
+
+Now we can see that ``docker://docker.io`` shows up under
+``Authenticated Logins`` and Singularity will automatically supply the
+configured credentials when interacting with DockerHub.
+
+We can log into multiple OCI registies at the same time:
```
Comment (dtrudg) [inline endpoint.rst]:
```suggestion
We can log into multiple OCI registries at the same time:
```

**Review by dtrudg — CHANGES_REQUESTED** [review]
Looks pretty good to me.

A few comments to take or leave.

Could you add a brief mention of the `remote login` authentication and a reference link also to the singularity_and_docker page? : https://github.com/hpcng/singularity-userdocs/blob/master/singularity_and_docker.rst#authentication-via-interactive-login

File: endpoint.rst (hunk 2 above)
Comment (ikaneshiro) [reply endpoint.rst:275]: Created in https://github.com/hpcng/singularity-admindocs/pull/108

**Comment by ikaneshiro** [comment]
@dctrud Good idea. Just added a reference in https://github.com/hpcng/singularity-userdocs/blob/master/singularity_and_docker.rst to help clarify.

**Review by dtrudg — APPROVED** [review]

**Comment by dtrudg** [comment]
@ikaneshiro LGTM - feel free to merge unless you want cclerget to review also.

**Pull request merged**